src/code/z_parameter.c

```diff
@@ -52,38 +293,753 @@
+            } else {
+                BUTTON_ITEM_EQUIP(CUR_FORM, EQUIP_SLOT_B) =
+                    GET_CUR_EQUIP_VALUE(EQUIP_TYPE_SWORD) - EQUIP_VALUE_SWORD_KOKIRI + ITEM_SWORD_KOKIRI;
+            }
+        }
+    }
+
+    for (btn = EQUIP_SLOT_B; btn <= EQUIP_SLOT_B; btn++) {
+        if ((GET_CUR_FORM_BTN_ITEM(btn) != ITEM_NONE) && (GET_CUR_FORM_BTN_ITEM(btn) != ITEM_UNK_FD)) {
+            Interface_LoadItemIconImpl(play, btn);
+        }
+    }
+}
+
+s32 Inventory_HasEmptyBottle(void) {
+    s32 slot;
+
+    for (slot = SLOT_BOTTLE_1; slot <= SLOT_BOTTLE_6; slot++) {
+        if (gSaveContext.save.inventory.items[slot] == ITEM_BOTTLE) {
+            return true;
+        }
+    }
+    return false;
+}
+
+s32 Inventory_HasItemInBottle(u8 item) {
+    s32 slot;
+
+    for (slot = SLOT_BOTTLE_1; slot <= SLOT_BOTTLE_6; slot++) {
+        if (gSaveContext.save.inventory.items[slot] == item) {
+            return true;
+        }
+    }
+    return false;
+}
+
+void Inventory_UpdateBottleItem(PlayState* play, u8 item, u8 btn) {
+    gSaveContext.save.inventory.items[GET_CUR_FORM_BTN_SLOT(btn)] = item;
+    SET_CUR_FORM_BTN_ITEM(btn, item);
+
+    Interface_LoadItemIconImpl(play, btn);
+
+    play->pauseCtx.cursorItem[PAUSE_0] = item;
+    gSaveContext.buttonStatus[btn] = BTN_ENABLED;
+
+    if (item == ITEM_HOT_SPRING_WATER) {
+        func_8010EBA0(60, GET_CUR_FORM_BTN_SLOT(btn) - SLOT_BOTTLE_1);
+    }
+}
+
+s32 Inventory_ConsumeFairy(PlayState* play) {
+    u8 bottleSlot = SLOT(ITEM_FAIRY);
+    u8 btn;
+    u8 i;
+
+    for (i = BOTTLE_FIRST; i < BOTTLE_MAX; i++) {
+        if (gSaveContext.save.inventory.items[bottleSlot + i] == ITEM_FAIRY) {
+            for (btn = EQUIP_SLOT_C_LEFT; btn <= EQUIP_SLOT_C_RIGHT; btn++) {
+                if (GET_CUR_FORM_BTN_ITEM(btn) == ITEM_FAIRY) {
+                    SET_CUR_FORM_BTN_ITEM(btn, ITEM_BOTTLE);
+                    Interface_LoadItemIconImpl(play, btn);
+                    bottleSlot = GET_CUR_FORM_BTN_SLOT(btn);
+                    i = 0;
+                    break;
+                }
+            }
+            gSaveContext.save.inventory.items[bottleSlot + i] = ITEM_BOTTLE;
+            return true;
+        }
+    }
+
+    return false;
+}
+
+/**
+ * Only used to equip Spring Water when Hot Spring Water timer runs out.
+ */
+void Inventory_UpdateItem(PlayState* play, s16 slot, s16 item) {
+    s16 btn;
+
+    gSaveContext.save.inventory.items[slot] = item;
+
+    for (btn = EQUIP_SLOT_C_LEFT; btn <= EQUIP_SLOT_C_RIGHT; btn++) {
+        if (GET_CUR_FORM_BTN_SLOT(btn) == slot) {
+            SET_CUR_FORM_BTN_ITEM(btn, item);
+            Interface_LoadItemIconImpl(play, btn);
+        }
+    }
+}
 
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_parameter/func_801153C8.s")
 
+TexturePtr sDoActionTextures[] = {
+    gDoActionAttackENGTex,
+    gDoActionCheckENGTex,
+};
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_parameter/func_80115428.s")
```

Comment:
Same with space here.

Reply:
I think spacing when the specific function is decompiled is fine, but I think having no-space when it's still `#pragma` just helps group data to the function it's attached to. Thoughts?

Reply:
I don't think it really matters, I just prefer to not have data right above functions, even if pragmas.